app/src/main/java/com/termux/app/TermuxActivity.java

```diff
@@ -198,6 +199,20 @@ public final class TermuxActivity extends AppCompatActivity implements ServiceCo
     @Override
     public void onCreate(Bundle savedInstanceState) {
         Logger.logDebug(LOG_TAG, "onCreate");
+
+        // For details see:
```

Comment:
TermuxService also requests the permission if its required to start foreground terminal session from background. Permissions must not be requested unless needed.

https://github.com/termux/termux-app/blob/8e3a8980a849046adbd4156741e1d84047ee1df6/app/src/main/java/com/termux/app/TermuxService.java#L717